app/assets/javascripts/workarea/storefront/flow_io/flow.js

```diff
@@ -17,75 +17,66 @@ WORKAREA.registerModule('flow', (function () {
-            } else {
-                return url.replace(oldLocale, locale);
-            }
-        },
-
-        updateSession = function(status, session) {
-            var country = session.geo.country.iso_3166_3.toLowerCase(),
-                url = WORKAREA.url.current();
-
-            WORKAREA.cookie.create('flow_country', country, 365);
-            WORKAREA.cookie.create('flow_experience', encodeURIComponent(JSON.stringify(session.experience)), 365);
-            WORKAREA.url.redirectTo(redirectUrl(url, country));
-        },
-
         initCountryPicker = _.once(function() {
             flow.cmd('on', 'ready', function () {
                 flow.countryPicker.createCountryPicker({
                     type: "modal",
-                    containerId: "country-picker",
-                    onSessionUpdate: updateSession
+                    containerId: "country-picker"
                 });
             });
         }),
 
-        localizePrices = function() {
-            flow.cmd('localize');
+        localizePrice = function(index, priceWrapper, experienceKey, currencyCode) {
+            var $priceWrapper = $(priceWrapper),
+                localPrices = _.merge({}, $priceWrapper.data('localPrices')),
+                priceKey = experienceKey + "-" + currencyCode,
+                priceLabel = localPrices[priceKey],
+                priceAttr;
+
+            if (_.isUndefined(priceLabel)) {
+              priceAttr = $priceWrapper.data('flowPriceAttr');
+              if (_.isUndefined(priceAttr)) {
+                $priceWrapper.attr('data-flow-localize', 'item-price');
+              }
+              else {
+                $priceWrapper.attr('data-flow-localize', 'item-price-attribute');
+                $priceWrapper.attr('data-flow-item-price-attribute', priceAttr);
+              }
+            }
+            else{
+              $priceWrapper.removeAttr('data-flow-localize');
+              $priceWrapper.removeAttr('data-flow-item-price-attribute');
+              $priceWrapper.html(priceLabel);
+           }
+        },
+
+        localizePrices = function($scope) {
+            flow.cmd('on', 'ready', function () {
+                var experienceKey = flow.session.getExperience(),
+                    currencyCode = flow.session.getCurrency();
+
+                $('[data-local-prices]', $scope).each(_.partialRight(localizePrice, experienceKey, currencyCode));
+                flow.cmd('localize');
+            });
         },
 
         /**
          * @method
          * @name init
          * @memberof WORKAREA.flow
          */
-        init = function () {
+        init = function ($scope) {
             if (_.isUndefined(WORKAREA.config.flow)) { return; }
 
             getScript();
-            localizePrices();
+            localizePrices($scope);
 
-            if (!_.isEmpty($("#country-picker"))) {
+            if (!_.isEmpty($("#country-picker", $scope))) {
```

Comment:
country picker isn't scoped, its a global html id

Reply:
I don't necessarily want to re-init the country picker if the `$scope` isn't `document`, though...

Reply:
Exactly, so $('#country-picker') won't be empty after the initial module init, but $('#country-picker', NEW_PDP_DOM) will always be empty

Reply:
oh that's true